u4-databases/features/uni-controller.js

```diff
@@ -0,0 +1,54 @@
+const { db } = require('../_services/firebase-initialized')
+
+const iterationList = async (req, res) => {
+  const query = db.collection('iterations').get()
+  const iterations = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-iteration-list', { iterations })
+}
+
+const iterationDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('iterations').doc(code).get()
+  const iteration = (await query).data()
+
+  res.render('uni-iteration-details', { iteration })
+}
+
+const studentDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('students').doc(code).get()
+  const student = (await query).data()
+
+  res.render('uni-student-details', { student })
+}
+
+const courseList = async (req, res) => {
+  const query = db.collection('iterations')
+    .where('year', '==', 2021)
+    .where('semester', '==', 1)
+    .get()
+  const courses = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-course-list', { courses })
+}
+
+const courseDetails = async (req, res) => {
+  const iterationCode = req.params.code
+  const query = db.collection('iterations')
+    .where('courseCode', '==', iterationCode)
+    .orderBy('iterationCode', 'desc')
+    .limit(1)
+    .get()
+  const iteration = (await query).docs[0].data()
```

Comment:
คุณอาจลองตั้งชื่อตัวแปรนี้ให้ละเอียดยิ่งขึ้น เช่น `latestIteration` หรือ `currentIteration`
You might consider giving this variable a more precise name, for example: `latestIteration` or `currentIteration`.